src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,117 @@
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts);
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) === selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
+            setViewpointFactList((prev) => [...prev, selectedFact]);
+
+            //remove the selected fact from the searchData
+            const newSearchData = searchData.filter(
+                (fact) => fact.id !== selectedFact.id,
+            );
+            setSearchData(newSearchData);
+            setSelectedFactId("");
+        }
+    }, [
+        selectedFactId,
+        setSelectedFactId,
+        viewpointFactList,
+        setViewpointFactList,
+        searchData,
+    ]);
+
+    const removeFact = (factId: string) => {
+        //remove the fact from the viewpointFactList
+        const removedFact = viewpointFactList.find(
+            (fact) => String(fact.id) == factId,
+        );
+        setViewpointFactList(
+            viewpointFactList.filter((fact) => String(fact.id) !== factId),
+        );
+
+        //add the fact back to the searchData
+        const newSearchData = [...searchData, removedFact].filter(
+            (fact): fact is Fact => fact !== undefined,
+        );
+        setSearchData(newSearchData);
+    };
+
+    return (
+        <div className="h-full rounded-lg bg-neutral-100 px-7 py-4">
+            <h1 className="mb-1 text-lg font-semibold text-neutral-700">
+                事實
+            </h1>
+            <div className="flex w-full items-center py-1 pr-[52px]">
+                <MagnifyingGlassIcon className="inline-block h-5 w-5 stroke-neutral-500" />
+                <Select
+                    variant="unstyled"
+                    searchable
+                    value={searchValue}
+                    onChange={(selectedFactId) => {
+                        setSelectedFactId(selectedFactId ? selectedFactId : "");
+                        console.log("User selected fact id:", selectedFactId);
+                    }}
+                    data={searchData.map((fact) => ({
+                        value: String(fact.id),
+                        label: fact.title,
+                    }))}
+                    checkIconPosition="right"
+                    radius={0}
+                    w="100%"
+                    limit={10}
+                    classNames={{
+                        input: "ml-2 bg-transparent text-lg font-normal text-neutral-500 focus-within:outline-b-2 focus-within:border-b-emerald-500 focus-within:outline-none",
+                    }}
+                    placeholder="搜尋 CommonGround"
+                    nothingFoundMessage="自行新增事實"
```

Comment:
This text should probably be "找不到想引著的事實嗎？將其引入 CommonGround 吧!", and should be a button that has the same effect as the import fact button below. It can be a button because, if you notice the type of "nothingFoundMessage", it actually expects a React Node, and thus means we can pass anything inside.